pkg/kubelet/kubelet.go

```diff
@@ -2617,33 +2629,41 @@ func (kl *Kubelet) setNodeStatus(node *api.Node) error {
 		}
 	}
 
-	updated := false
+	needUpdated := true
+	needRecordEvent := true
```

Comment:
We only need to record an event when there is a transition. I prefer setting the default to false, and flip the variable after line 2641. Also, rename the variable to `needToRecordEvent` or `shouldRecordEvent`.